### PR TITLE
[SofaNG] Re-organize unit tests 

### PR DESCRIPTION
**Pull request by fredroy**
Based on #2813 

and sofang-ize paths/plugins/etc.
and creation of some *Testing framework (solidmechanics, topology, mapping...)

and a loooots of clean as well. (not enough)

Some tests were disabled, they are back again

~~Will be updated when Collision/engine/etc will be merged.~~ done

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

File: Component/IO/Mesh/tests/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,20 @@
+cmake_minimum_required(VERSION 3.12)
+
+project(Sofa.Component.IO.Mesh_test)
+
+set(SOURCE_FILES
+    MeshExporter_test.cpp
+    MeshGmshLoader_test.cpp
+    MeshOBJLoader_test.cpp
+    # MeshSTLLoader_test.cpp
```
Comment (alxbilger) [inline Component/IO/Mesh/tests/CMakeLists.txt]: Why commented?
Either remove the file, or activate it.

**Review by alxbilger — CHANGES_REQUESTED** [review]
What a tremendous work!

I have only one comment. And you also have conflicts to solve.

**Review by guparan — CHANGES_REQUESTED** [review]
Correct me if I'm wrong but I think all the Testing packages should be using `sofa_create_package_with_targets` macro and providing a Config.cmake.in file.

**Comment by fredroy** [comment]
[ci-build][force-full-build][with-all-tests]

**Comment by fredroy** [comment]
[ci-build][force-full-build][with-all-tests]

**Pull request merged**